src/core/mesh/qgsmeshdataprovider.h

```diff
@@ -556,6 +556,11 @@ class CORE_EXPORT QgsMeshDataProvider: public QgsDataProvider, public QgsMeshDat
   signals:
     //! Emitted when some new dataset groups have been added
     void datasetGroupsAdded( int count );
+
+
+    //! Emitted when one or some extras uris can't be loaded. The mesh layer will receive this signal and will treat the rendere settings
```

Comment:
Can you just remove the parameter, since I do not thing that the logic you created is valid for all cases and I find it a bit confusing to understand. I suggest

```
//! Emitted when one or more dataset groups have been removed
void datasetGroupsRemoved();
```